### ISSUE TITLE
Meta schema platform versions use inconsistent identifiers across distributions

### ISSUE THREAD

**Issue by vicmunoz** [ansible/ansible-lint#5101]
##### Summary

The `galaxy_info.platforms[].versions` schema appears to use different version identifiers depending on the distribution family, which makes it difficult or unreliable to compare supported platforms declared in `meta/main.yml` with Ansible distribution facts.

For Debian/Ubuntu-based distributions, the schema expects release codenames, matching `ansible_distribution_release`, for example:

* Ubuntu: `focal`, `jammy`, `noble`, etc.
* Debian: `bullseye`, `bookworm`, etc.

For AlmaLinux, the schema expects major versions, matching `ansible_distribution_major_version`, for example:

* AlmaLinux: `9`, `10`

This creates an inconsistency when role logic attempts to validate the current host against the platforms declared in `meta/main.yml`, because the correct fact to compare against depends on the distribution.

##### Issue Type

* Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint --version
```

```console
ansible-lint 26.4.0 using ansible-core:2.16.15 ansible-compat:26.3.0 ruamel-yaml:0.19.0 ruamel-yaml-clib:0.2.15
```

* ansible installation method: uv
* ansible-lint installation method: uv

Relevant target OS facts:

Ubuntu example:

```json
{
  "ansible_distribution": "Ubuntu",
  "ansible_distribution_file_parsed": true,
  "ansible_distribution_file_path": "/etc/os-release",
  "ansible_distribution_file_variety": "Debian",
  "ansible_distribution_major_version": "26",
  "ansible_distribution_release": "resolute",
  "ansible_distribution_version": "26.04"
}
```

AlmaLinux example:

```json
{
  "ansible_distribution": "AlmaLinux",
  "ansible_distribution_file_parsed": true,
  "ansible_distribution_file_path": "/etc/redhat-release",
  "ansible_distribution_file_variety": "RedHat",
  "ansible_distribution_major_version": "10",
  "ansible_distribution_release": "Heliotrope Lion",
  "ansible_distribution_version": "10.1"
}
```

##### STEPS TO REPRODUCE

Create a role with the following `meta/main.yml`:

```yaml
galaxy_info:
  platforms:
    - name: Ubuntu
      versions:
        - focal
        - jammy
        - noble
        - resolute
    - name: AlmaLinux
      versions:
        - "9"
        - "10"
```

Then gather facts on supported target hosts and try to compare the declared platform version from `galaxy_info.platforms[].versions` with Ansible distribution facts.

For Ubuntu, the value declared in `meta/main.yml` matches:

```yaml
ansible_distribution_release
```

Example:

```yaml
ansible_distribution: Ubuntu
ansible_distribution_release: resolute
ansible_distribution_major_version: "26"
ansible_distribution_version: "26.04"
```

For AlmaLinux, the value declared in `meta/main.yml` matches:

```yaml
ansible_distribution_major_version
```

Example:

```yaml
ansible_distribution: AlmaLinux
ansible_distribution_release: Heliotrope Lion
ansible_distribution_major_version: "10"
ansible_distribution_version: "10.1"
```

This means generic validation logic cannot consistently compare `galaxy_info.platforms[].versions` against a single Ansible fact.

##### Desired Behavior

The schema should use a consistent versioning convention across distributions, or the expected mapping should be clearly documented.

Ideally, platform versions in `meta/main.yml` should consistently map to one of the standard Ansible distribution facts, for example one of:

```yaml
ansible_distribution_release
ansible_distribution_major_version
ansible_distribution_version
```

Alternatively, the documentation/schema should explicitly define which fact each supported platform version is expected to correspond to.

This would allow roles to reliably use `galaxy_info.platforms` as a source of truth for supported target operating systems.

##### Actual Behavior

The current schema appears to use mixed conventions:

* Ubuntu/Debian versions use release codenames, matching `ansible_distribution_release`.
* AlmaLinux versions use major versions, matching `ansible_distribution_major_version`.

Example schema values:

```yaml
Ubuntu:
  - focal
  - jammy
  - noble
  - resolute

AlmaLinux:
  - "9"
  - "10"
```

Example Ubuntu facts:

```json
{
  "ansible_distribution": "Ubuntu",
  "ansible_distribution_major_version": "26",
  "ansible_distribution_release": "resolute",
  "ansible_distribution_version": "26.04"
}
```

Example AlmaLinux facts:

```json
{
  "ansible_distribution": "AlmaLinux",
  "ansible_distribution_major_version": "10",
  "ansible_distribution_release": "Heliotrope Lion",
  "ansible_distribution_version": "10.1"
}
```

As a result, role logic that attempts to compare the supported platforms declared in `meta/main.yml` with gathered facts must implement distribution-specific mappings.

For example, this works for Ubuntu:

```yaml
galaxy_info.platforms[].versions == ansible_distribution_release
```

But for AlmaLinux, the equivalent comparison must be:

```yaml
galaxy_info.platforms[].versions == ansible_distribution_major_version
```

This inconsistency makes the schema hard to use programmatically and can lead to incorrect validation logic in roles.

**Comment by vicmunoz** [comment]
Escalated from https://github.com/ansible/ansible-lint/discussions/5092
